lib/mergeset/part_search.go

```diff
@@ -292,9 +317,8 @@ func (ps *partSearch) readIndexBlock(mr *metaindexRow) (*indexBlock, error) {
 	if err != nil {
 		return nil, fmt.Errorf("cannot decompress index block: %w", err)
 	}
-	idxb := &indexBlock{
-		buf: append([]byte{}, ps.indexBuf...),
-	}
+	idxb := ps.tmpIdB
+	idxb.buf = append(idxb.buf[:0], ps.indexBuf...)
```

Comment:
Nit: consider moving the reset logic into `indexBlock.reset()` to make the intent clearer and the logic more explicit.

Reply:
I mean something like this:
```
idxb := ps.tmpIdB
idxb.reset();
idxb.buf = append(idxb.buf, ps.indexBuf...)
idxb.bhs, err = unmarshalBlockHeadersNoCopy(idxb.bhs, idxb.buf, int(mr.blockHeadersCount))
```

It aligns with the code here https://github.com/VictoriaMetrics/VictoriaMetrics/pull/9324/files#diff-1de24c0798b2bd46d71137d276af8122f111b0117ee91228368da1f4a58260d8R364 and makes reset thing explicit.